reactor-support/src/test/java/tech/picnic/reactor/ReactorSpring4UtilTest.java

```diff
@@ -0,0 +1,232 @@
+package tech.picnic.reactor;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+import static org.springframework.http.HttpStatus.OK;
+import static tech.picnic.reactor.ReactorSpring4Util.publisherToDeferredResult;
+import static tech.picnic.reactor.ReactorSpring4Util.publisherToSse;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.google.common.collect.ImmutableList;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.TestInstance.Lifecycle;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.converter.StringHttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.http.converter.xml.MappingJackson2XmlHttpMessageConverter;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.ResultActions;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.context.request.async.DeferredResult;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.test.scheduler.VirtualTimeScheduler;
+
+@TestInstance(Lifecycle.PER_CLASS)
+class ReactorSpring4UtilTest {
+  private VirtualTimeScheduler testScheduler;
+
+  @SuppressWarnings("NullAway")
+  private MockMvc mockMvc;
+
+  @BeforeEach
```

Comment:
This is different than the rx test setup in the sense that we do this each time. Changing this to `@BeforeAll` means the scheduler is re-used between test runs and gives unexpected results. Most notably setting the time to 0 (closest thing to a reset) can also trigger responses sent